src/main/java/com/puppycrawl/tools/checkstyle/Checker.java

```diff
@@ -108,7 +108,7 @@ public class Checker extends AbstractAutomaticBean implements MessageDispatcher,
     private Context childContext;
 
     /** The file extensions that are accepted. */
-    private String[] fileExtensions = CommonUtil.EMPTY_STRING_ARRAY;
+    private String[] fileExtensions;
```

Comment:
Null is ok by this utility
https://github.com/checkstyle/checkstyle/blob/223ed3981ebf088c005dc7c8bf2d075194f77ae1/src/main/java/com/puppycrawl/tools/checkstyle/utils/CommonUtil.java#L120-L124